scripts/build_test.py

```diff
@@ -101,3 +366,95 @@ def test_get_hashes_json_file_contents(self):
+        self.assertEqual(total_file_count - 1, len(build_tasks))
+        build_tasks.clear()
+
+        # Only build HTML files.
+        build.build_files(
+            build.ASSETS_DEV_DIR, build.ASSETS_OUT_DIR, asset_hashes,
+            build_tasks, file_formats=('.html',))
+        total_html_file_count = 0
+        for _, _, files in os.walk(build.ASSETS_DEV_DIR):
+            for filename in files:
+                if filename.endswith('.html'):
+                    total_html_file_count += 1
+        self.assertEqual(total_html_file_count, len(build_tasks))
+
+    def test_rebuild_new_files(self):
+        """Test rebuid_new_files queue up a corresponding number of build tasks
+        to the number of file changes.
+        """
+        build_tasks = collections.deque()
+        new_file_name = 'manifest.json'
+        recently_changed_filenames = [
+            os.path.join(build.ASSETS_DEV_DIR, 'i18n', new_file_name)]
+        asset_hashes = build.get_file_hashes(build.ASSETS_DEV_DIR)
+        build.rebuild_new_files(
+            build.ASSETS_DEV_DIR, build.ASSETS_OUT_DIR,
+            recently_changed_filenames, asset_hashes, build_tasks)
+        self.assertEqual(len(build_tasks), len(recently_changed_filenames))
+
+    def test_get_recently_changed_filenames(self):
+        """Test get_recently_changed_filenames to detect file recently added."""
+        # Obtain hashes from DEV's /assets.
+        assets_hashes = build.get_file_hashes(build.ASSETS_DEV_DIR)
+        # Create an empty sandbox folder, to simulate an empty /build folder.
+        SANDBOX_DIR = os.path.join(build.PARENT_DIR, 'sandbox')
+        build.ensure_directory_exists(SANDBOX_DIR)
+        recently_changed_filenames = []
+        # Assert that list is init to 0.
+        self.assertEqual(len(recently_changed_filenames), 0)
+        recently_changed_filenames = build.get_recently_changed_filenames(
+            assets_hashes, SANDBOX_DIR)
+        # Since all HTML and Python files are already built, they are ignored.
+        with self.swap(build, 'FILE_EXTENSIONS_TO_IGNORE', ('.html', '.py',)):
+            self.assertEqual(
+                len(recently_changed_filenames), build.get_file_count(
+                    build.ASSETS_DEV_DIR) - 1)
+        # Minus 1 from ASSETS_DEV_DIR to account for added hashes.js.
+
+    def test_remove_deleted_files(self):
+        """Test remove_deleted_files to queue up the correct number of deletion
+        task.
+        """
+        delete_tasks = collections.deque()
+        # Using empty dict, indicating an empty directory,
+        # which means that all files should be removed.
+        file_hashes = dict()
+        # Assert that queue is init to 0.
+        self.assertEqual(len(delete_tasks), 0)
+        build.remove_deleted_files(
+            file_hashes, build.THIRD_PARTY_GENERATED_DEV_DIR, delete_tasks)
+        # Assert that all files from /assets should be queued up for deletion.
+        self.assertEquals(
```

Comment:
Hm, shouldn't this be assertEqual?

Also does this mean your backend tests are not actually picked up by unittest when the test suite is run on Travis? (All tests are passing...)

Reply:
Well, `build_test` passed every build up to this point:
https://travis-ci.org/oppia/oppia/jobs/417477129#L681

Anyhow, looks like I will replace `assertEquals` with `assertEqual` since it is deprecated.

Reply:
FYI, a lot of these `assertEquals` were already in place from #3454.

Reply:
Ah. @apb7 can we fix this and update the linter to guard against it please?

Reply:
Yep, I just did a `git grep assertEquals` and seem like it is quite prevalent. Looks likes the deprecation was quite recent.